x-pack/filebeat/input/streaming/config.go

```diff
@@ -43,6 +47,12 @@ type redact struct {
 	Delete bool `config:"delete"`
 }
 
+type retry struct {
+	MaxAttempts *int           `config:"max_attempts"`
+	WaitMin     *time.Duration `config:"wait_min"`
+	WaitMax     *time.Duration `config:"wait_max"`
+}
+
```

Comment:
[Suggestion]
Do no use pointers here, the `retry` struct is already a pointer in `config`, so if it is not `nil`, then retry should be used.

This will make the validation simpler, `MaxAttempts` should be required to be `>0`, `WaitMin` and `WaitMax` should also be required to be `>0s`

Reply:
This will also make the validation simpler because you can just set the struct tags for the validation of the fields and let the config library perform the validation.